x-pack/plugins/fleet/server/routes/agent_policy/handlers.ts

```diff
@@ -386,7 +386,10 @@ export const getFullAgentPolicy: FleetRequestHandler<
       const fullAgentPolicy = await agentPolicyService.getFullAgentPolicy(
         soClient,
         request.params.agentPolicyId,
-        { standalone: request.query.standalone === true }
+        {
+          standalone: request.query.standalone === true,
+          standalone_api_key: request.query.standalone_api_key,
```

Comment:
I think there is a potential security issue to have the api_key in the query, `standalone_api_key` could become visible in some logs, I am wondering if we should have a POST version of that endpoint where we can pass this in the body

Reply:
Eek, you're right! I think there are 3 approaches we can take here:
1. Not update the yaml at all and let the user edit it. It might be enough, I was just trying to go the extra mile by updating the yaml (like is done in the Logs onboarding).
2. Only edit it on the frontend side (this is what was done in the first commit of this PR). This means that it appears edited but when you download the policy then it still has the placeholder. I didn't like that option because it seemed inconsistent.
3. Change this implementation to retrieve the edited policy in a secure way. I'm not sure right now if that would be worth the risk/effort.

Reply:
I like the idea of doing it on the front-end side, and we could also change the download button to not be a link but generate the file client side, we already do this for the download Elastic JSON https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_details_page/components/agent_details_json_flyout.tsx#L38-L43 wdyt?

Reply:
> we could also change the download button to not be a link but generate the file client side
@nchaulet I took a look to implement it but this would involve having part of the [getFullAgentPolicy](https://github.com/elastic/kibana/blob/6394e46e17f43bae68c9b42beed10863571498d6/x-pack/plugins/fleet/server/services/agent_policies/full_agent_policy.ts#L67) logic in the frontend?

Unless I'm missing something, I'm not sure how we could generate the file directly on the client side; for now I'm going to remove the `standalone_api_key` from the query and fallback on this solution proposed by Jill, which seems to me the safer and shorter. 
> Only edit it on the frontend side (this is what was done in the first commit of this PR). This means that it appears edited but when you download the policy then it still has the placeholder.

Happy to do further changes! We also have the option of having an enhancement ticket if this is solution is not ideal.

Reply:
I pushed https://github.com/elastic/kibana/pull/187133/commits/c59a3e34009c72729ed4aaa71bba31128dcd2cb4 where I removed `standalone_api_key`. With this change, the `API_KEY` is not updated in the YAML:
![Uploading Screenshot 2024-07-01 at 12.59.13.png…]()